libyul/optimiser/StructuralSimplifier.h

```diff
@@ -31,6 +31,7 @@ namespace yul
  * - turn switch with single case into if
  * - replace switch with only default case with pop(expression) and body
  * - replace switch with const expr with matching case body
+ * - remove cases with empty bodies from switches (if no non-empty default case exists)
  * - remove for with false condition
```

Comment:
This should now include the switch-with-no-cases to pop(expression) step.

Reply:
To me it feels like otherwise having to document it here alone (like "replaces with pop(expression), if expression is not a literal and removes it entirely otherwise" or something) is another point in favour of just keeping all the ``pop``s and letting them be removed elsewhere.